### PR TITLE
Information packet

### LINKED ISSUE #213: [Feature Request] New packet type for Session/server information
Currently, when joining a private server, no details about the server are displayed. When joining a custom server, details like an extended description, custom data from scripts, etc don't exist, yet.

- Add a new packet type for session information
- Add subtypes for extended description, mod-provided info?

### PR DESCRIPTION
**Pull request by WiserTixx**
This PR adds an "I" packet which returns the server information. This can be used by external programs and the launcher to get information about a server without having to connect it to. It can be toggled in the config and in lua.

### CODE REVIEW

File: src/THeartbeatThread.cpp [hunk 1]
```diff
@@ -138,25 +139,30 @@ void THeartbeatThread::operator()() {
...
+        {"clientversion", Application::ClientMinimumVersion().AsString()},
+        {"name", Application::Settings.getAsString(Settings::Key::General_Name)},
+        {"tags", Application::Settings.getAsString(Settings::Key::General_Tags)},
+        {"guests", Application::Settings.getAsBool(Settings::Key::General_AllowGuests)},
+        {"modlist", mResourceManager.TrimmedList()},
+        {"modstotalsize", mResourceManager.MaxModSize()},
+        {"modstotal", mResourceManager.ModsLoaded()},
+        {"playerslist", GetPlayers()},
+        {"desc", Application::Settings.getAsString(Settings::Key::General_Description)}
+    };
```
Comment (lionkor) [inline src/THeartbeatThread.cpp]: please format with clang-format with the .clang-format file in the repo

File: src/THeartbeatThread.cpp [hunk 2]
```diff
@@ -65,7 +66,7 @@ void THeartbeatThread::operator()() {
         json::Document Doc;
         bool Ok = false;
         for (const auto& Url : Application::GetBackendUrlsInOrder()) {
-            T = Http::POST(Url, 443, Target, Body, "application/x-www-form-urlencoded", &ResponseCode, { { "api-v", "2" } });
+            T = Http::POST(Url, 443, Target, Body, "application/json", &ResponseCode, { { "api-v", "2" } });
```
Comment (lionkor) [inline src/THeartbeatThread.cpp:69]: I assume the backend already supports this? Very nice :)

**Review by lionkor — CHANGES_REQUESTED** [review]
very nice, just formatting :)

File: src/THeartbeatThread.cpp (hunk 2 above)
Comment (WiserTixx) [reply src/THeartbeatThread.cpp:69]: It seems like it always has lol

Comment (lionkor) [reply src/THeartbeatThread.cpp:69]: fucking hell

**Review by lionkor — APPROVED** [review]

**Pull request merged**